### PR TITLE
fix: adds prefix length error to eos cli plugin

### LINKED ISSUE #57773: Ansilble arista eos_config not working as expected
##### SUMMARY

ansible 2.7.7

##### ISSUE TYPE
Trying to push Arista configuration commands using ansible src file.
After running the playbook its successfully executed however some commands are not configured on actual device which are there is ansible src config file.

##### COMPONENT NAME
Ansible eos_config module for Arista

##### ADDITIONAL INFORMATION
So ansible src file has following command:
```
interface Ethernet3/11/1
   no switchport
   ip address 1.1.1.1/32
```
However when i login to the actual device it gives me following error:
```
(s1)(config)#int Ethernet3/11/1
(s1)(config-if-Et3/11/1)#no switchport
(s1)(config-if-Et3/11/1)#ip address 1.1.1.1/32
% Prefix length must be less than 32
PLAY RECAP *************************************************************************************************************************************************************************
S1                : ok=1    changed=1    unreachable=0    failed=0   
```
Just need help to determine why ansible is not reporting failed error instead of saying successful

more arista_config_apply.yaml
---
```
- name: Arista config apply
  hosts: Arista
  gather_facts: false
  vars:
    ansible_connection: network_cli
    ansible_network_os: eos
    ansible_become: yes
    authorize: yes
    ansible_become_method: enable
  ignore_errors: no
  #vars:
   # cli:
   #   transport: cli
   #   authorize: true

  tasks:
    - name: Apply config
      eos_config:
        diff_against: running
        src: /etc/ansible/CONFIGS/Temp.cfg
```

This is config file Temp.cfg

```
interface Ethernet 3/13/1 
   no shutdown
!
interface Ethernet3/11/1
   no switchport
   ip address 1.1.1.1/32 ============? Expecting error in Ansible however its success.

!
interface Ethernet3/21/1
   no switchport
   ip address 3.3.3.1/30
!
interface loopback100
   description ***test interface ***
   ip address 1.1.1.1/32
!
interface Port-Channel 503
   description test
   switchport trunk allowed vlan none
   switchport mode trunk
   spanning-tree portfast
!
interface Port-Channel 504
   description test
   switchport trunk allowed vlan none
   switchport mode trunk

```

### PR DESCRIPTION
**Pull request by ghost**
##### SUMMARY

Adding terminal error for prefix length that currently passes, breaking idempotency.

Fixes: #57773
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

eos_config
##### ADDITIONAL INFORMATION

eos error presents as below when attempting to input an invalid prefix length on an interface; this currently passes as a 'success' even when proposed config not applied.

```
$ docker exec -it ceos1 Cli

localhost#show int des
Interface                      Status         Protocol           Description
Et1                            up             up                 

localhost#show run int Et1
interface Ethernet1

localhost#conf
localhost(config)#int et1
localhost(config-if-Et1)#no switchport 
localhost(config-if-Et1)#ip address 10.1.1.1/32
% Prefix length must be less than 32
localhost(config-if-Et1)#ipv6 address 2404::1/128
% Prefix length must be less than 128
```

Playbook:
```
$ cat ceos_test.yaml 
---


- name: test application of config with invalid prefix
  hosts: ceos
  connection: network_cli

  tasks:
    - name: Apply config
      eos_config:
        diff_against: running
        lines:
          - no switchport
          - ip address 1.1.1.1/32
        parents:
          interface Ethernet1
```

Initial application of this playbook succeeds:
```
$ ansible-playbook ceos_test.yaml -i inventory-docker.py -u admin -k
SSH password: 

PLAY [test application of config with invalid prefix] ****************************************************************************************************************************************

TASK [Apply config] **************************************************************************************************************************************************************************
changed: [ceos]

PLAY RECAP ***********************************************************************************************************************************************************************************
ceos      : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

However the config does not apply:
```
$ docker exec -it ceos Cli
localhost#show run int Ethernet1
interface Ethernet1
   no switchport
```

Repeating the playbook demonstrates its lack of idempotency:

```
[cloudbuilders@WSQGPQF2 cloudbuilders]$ ansible-playbook ceos_test.yaml -i inventory-docker.py -u admin -k
SSH password: 

PLAY [test application of config with invalid prefix] ****************************************************************************************************************************************

TASK [Apply config] **************************************************************************************************************************************************************************
changed: [ceos]

PLAY RECAP ***********************************************************************************************************************************************************************************
test-c99-f0102-ltr01a      : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

Post-change, a failure is seen in line with other invalid configuration application:
```
$ ansible-playbook ceos_test.yaml -i inventory-docker.py -u admin -k
SSH password: 

PLAY [test application of config with invalid prefix] ****************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************************
ok: [ceos]

TASK [Apply config] **************************************************************************************************************************************************************************
fatal: [ceos]: FAILED! => {"changed": false, "data": "ip address 1.1.1.1/32\r\n% Prefix length must be less than 32\r\nlocalhost(config-s-ansibl-if-Et1)#", "msg": "ip address 1.1.1.1/32\r\n% Prefix length must be less than 32\r\nlocalhost(config-s-ansibl-if-Et1)#"}

PLAY RECAP ***********************************************************************************************************************************************************************************
ceos      : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/205

### CODE REVIEW

File: lib/ansible/plugins/terminal/eos.py [hunk 1]
```diff
@@ -49,6 +49,8 @@ class TerminalModule(TerminalBase):
         re.compile(br"% More than \d+ OSPF instance", re.I),
         re.compile(br"% Subnet [0-9a-f.:/]+ overlaps", re.I),
         re.compile(br"Maximum number of pending sessions has been reached"),
+        # match v4 & v6 prefix length errors
+        re.compile(br"% Prefix length must be less than (?:32|128)"),
```
Comment (Qalthos) [inline lib/ansible/plugins/terminal/eos.py]:
```suggestion
        re.compile(br"% Prefix length must be less than"),
```
We can almost certainly get away with being slightly less precise with this error.

**Review by Qalthos — CHANGES_REQUESTED** [review]

**Comment by pabelanger** [comment]
recheck

File: lib/ansible/plugins/terminal/eos.py (hunk 1 above)
Comment (ghost) [reply lib/ansible/plugins/terminal/eos.py]: Agreed, committed.

**Review by Qalthos — APPROVED** [review]

**Comment by Qalthos** [comment]
recheck

**Pull request merged**